### DISCUSSION TITLE
1 seule modale appelée depuis plusieurs acteurs

### DISCUSSION THREAD

**Discussion by edejean** [witsa/synapps#596] [General]
Bonjour,
Je cherche à appeler une seule modale depuis plusieurs acteurs.
Une partie de cette modale pointera sur des variables fixes, l'autre partie affichera du contenu relatif à l'acteur utilisé.
J'ai plusieurs k7 de clim avec réglages généraux et spécifiques à la K7.

L'équivalent de ce que je faisais en synoptique avant synaps...

var String MyZone = ":System.Attribute.Zone.BARRIERES."

 MyZone = ":System.Attribute.Zone.BARRIERES."&&.ActorValue
 if OnDrawBack then
   ..ACT_1.Value = Value(MyZone&&".Title")Evaluation
   ..ACT_2.Param.Zone = MyZone
 end

**Comment by WITNicolas** [comment]
Bonjour,

Pour appeler une modale il faut un evenement OnClick sur les acteurs qui vont l'appeler.
Dans cet evenement il faudra saisir : 
`context.getActor('LabelModale').properties.isShown = true;`

Un exemple parmi tant d'autres pour passer un argument, il faudrait créer une additionnelle texte dans la modale (propChemin)
Et du coup compléter les evenements OnClick avec :
`context.getActor('LabelModale').additionals.propChemin= "information que je veux transmettre à la modale";`

ça devrait fonctionner.

cordialement,

**Reply by edejean** [reply]
Merci Nicolas, 

Nickel pour le OnClick, nickel pour récupérer la variable définie dans l'acteur qui sera le numéro de k7 mais là où je patoge légèrement, c'est comment inclure cette variable dans le type fournisseur de variable...

J'ai tenté çà, mais la variable n'est pas utilisée, j'ai vu que dans les additionnelles, j'avais le type "Chemin de variable" mais je ne sais pas l'utiliser....

![image](https://github.com/witsa/synapps/assets/105919791/60688acf-50fb-447b-a67c-054548ec743b)

![image](https://github.com/witsa/synapps/assets/105919791/43bc2f62-91ad-4800-92ea-c4dfee0f65eb)

**Reply by WITNicolas** [reply]
Effectivement, c'est un peu plus compliqué, mais tu n'es pas loin de la bonne pratique.

Pour concaténer une chaine pour en faire une source de donnée, il faut : 
Un fournisseur de variable qui va donner le chemin de base, c'est à dire : :easy.RESS.CVC.DRV_BUR
Un founisseur de variable relative avec comme clé parent le fournisseur précédent et en chemin relatif, le lien vers l'aditionnelle qui donnera la 2eme partie de la concaténation.

Aprés, tous tes acteurs seront à lier au fournisseur de variable relative qui sera donc le résultat de toute la concaténation.
Donc en changeant l'additionnelle, cela changera le chemin, donc la référence de tous les acteurs.

En fait tu es passé par un joker dans le chemin et pour le coup je ne suis pas sûr cela fonctionne convenablement. Il n'y a qu'un developpeur qui pourra te répondre précisement, en tout cas, la bonne pratique est l'enchainement en cascade d'un fournisseur de variable avec un fournisseur de variable relative.

**Reply by edejean** [reply]
Merci Nicolas,
Bon y-a-t-il un développeur qui pourrait me répondre du coup ? ;)

**Reply by Raphwit** [reply]
Effectivement, un joker au sein du chemin dans un liaison ne fonctionne pas, en revanche la solution proposée par @WITNicolas correspond parfaitement à la demande.

Il faut en effet utiliser un acteur fournisseur de variable et un acteur fournisseur de variable relatif qui prend le premier en acteur parent et une liaison vers une additionnelle de type chemin de variable dans son chemin relatif.

Par la suite, nous viendrons donc changer la valeur de cette additionnelle pour que l'acteur fournisseur de variable relatif se mette à jour automatiquement vers la nouvelle variable renseignée.

Pour se faire, dans le même évènement onClick que vous avez créée, vous pouvez changer la valeur de l'additionnelle en allant chercher tout d'abord l'acteur puis en changeant la valeur de sont additionnelle (créée préalablement) : 

`context.getActor('Nom_De_Mon_Acteur_Fournisseur_Relatif').additionals.Nom_De_Mon_Additionnelle = Nouvelle_valeur`

Ainsi, lors de l'ouverture de la modale, la valeur de l'additionnelle sera changée et correctement définie.

**Reply by edejean** [reply]
" et une liaison vers une additionnelle de type chemin de variable dans son chemin relatif. " ==> désolé mais ça, je n'arrive pas à le faire...

Dans ma modale, ajout de l'acteur fournisseur de variable :

![image](https://github.com/witsa/synapps/assets/105919791/dc85aadb-36bf-41c3-8b17-49df781f981b)

Dans ma modale, ajout de l'acteur fournisseur de variable relatif : c'est là où je bloque...

![image](https://github.com/witsa/synapps/assets/105919791/ad6046f5-904a-4765-aafe-7c30fe49f4ce)

Dans ma modale, un acteur texte qui devrait pointer sur le nom de ma RSS de K7_01 :

![image](https://github.com/witsa/synapps/assets/105919791/1fd6fed6-1d9e-4d44-8b4c-fd83e8109e44)

Mon chemin relatif n'est pas pris en compte :

![image](https://github.com/witsa/synapps/assets/105919791/b24e5939-b830-45e5-89e5-d140c4f257dd)

**Reply by Raphwit** [reply]
Si vous souhaitez que l'acteur texte pointe vers le nom de la ressource trouvée par l'acteur 'fournisseur de variable relatif', il faut simplement effectuer une liaison de type 'Fournisseur de variable' qui pointe sur le champ nom

@edejean la réponse à été modifiée

**Reply by manufitoussiwit** [reply]
![image](https://github.com/witsa/synapps/assets/35595723/9b92dd16-cea9-45b1-8e67-2101a6bd382d)
